CONTRIBUTING.md

```diff
@@ -13,6 +13,59 @@ This project and everyone participating in it is governed by the
 By participating, you are expected to uphold this code.
 Please report unacceptable behavior to the nmstate team.
 
+## Code structure
+
+The repository is structured as follows:
+
+- `./automation/` - Contains the [automation enviroment](./automation/README.md), 
+serving the tests of Nmstate.
+
+- `./doc/` - Contains the documentation. 
+
+- `./examples/` - Contains YAML examples for different configurations.
+
+- `./libnmstate/ifaces/` Contains the classes including the attributes of the 
+interfaces.
+ 
+- `./libnmstate/nispor/` Contains tools to comunicate with nispor. 
+
+- `./libnmstate/nm/` Contains the support of NetworkManager. 
```

Comment:
Contains the nmstate NetworkManager plugin, currently for querying network status and applying network states.